sqlmesh/magics.py

```diff
@@ -24,48 +28,52 @@
 from sqlmesh.utils import sqlglot_dialects, yaml
 from sqlmesh.utils.errors import MagicError, MissingContextException, SQLMeshError
 
+logger = logging.getLogger(__name__)
+
 CONTEXT_VARIABLE_NAMES = [
     "context",
     "ctx",
     "sqlmesh",
 ]
 
 
+def pass_sqlmesh_context(func: t.Callable) -> t.Callable:
```

Comment:
why was this added?

Reply:
Allows magics that need a context to access a context while ensuring it is configured properly. It kind of already existed but with a property but now this ensure you only get the context once and we don't have to implement any caching.